### ISSUE TITLE
`DirectoryMonitor` behavior on file rename can be surprising 

### ISSUE THREAD

**Issue by ranweiler** [microsoft/onefuzz#1858]
The following is arguably not a bug, but is a subtlety that should at least be documented.

Suppose we have: 
- An outer directory `outer` and a child directory `outer/watched`
- A `DirectoryMonitor` that is watching the child dir, `outer/watched`
- No other files or directories

The following shell transcript will result in the monitor reporting the creation of 1 file, `myfile.txt`:
```bash
cd outer

# Create `myfile.txt` in `outer`.
touch myfile.txt

# Move the file into the watched directory.
# 
# This is reported by the monitor.
mv myfile.txt ./watched/myfile.txt
```

Counterintuitively, from a clean slate, the following shell transcript will result in the monitor reporting the creation of 1 file, `.tmp.txt`.
```bash
cd outer/watched

# This is reported by the monitor.
touch .tmp.txt

# This is NOT reported as a created file.
mv .tmp.txt myfile.txt
``` 
This is true on both Linux and Windows. 

AB#35926

**Comment by mgreisen** [comment]
@ranweiler is this still true? Did you change the DirectoryMonitor?

**Comment by ranweiler** [comment]
I believe this is still true.

To change this, we'd add a new case [here](https://github.com/microsoft/onefuzz/blob/b38ab5de9ee206b5b442e249bddc05130193e9e4/src/agent/onefuzz/src/monitor.rs#L140). I suspect that we just need to watch for a [`Modify`](https://github.com/microsoft/onefuzz/blob/b38ab5de9ee206b5b442e249bddc05130193e9e4/src/agent/onefuzz/src/monitor.rs#L140) event with `ModifyKind` of [`Name`](https://docs.rs/notify/latest/notify/event/enum.ModifyKind.html#variant.Name).

**Comment by Porges** [comment]
This has been fixed in #3195. Renames inside the monitored directory now produce a notification.
